languages/elixir/exercises/concept/maps/.docs/after.md

```diff
@@ -0,0 +1,73 @@
+Great work, knowing how to effectivly use [maps][maps] is an import skill to use Elixir fluently.
+
+## Key points
+
+- [Maps][maps] are a data structure that holds key-value pairs.
+- Keys can be of any type, but must be unique.
+- Values can be of any type, they do not have to be unique.
+- [Maps][maps] do not guarantee the order of their contents despite appearing so.
+  - Their underlying implementation gives this misperception:
+    - At small sized (<=32 entries), they are implemented as an ordered [Keyword list][keyword-list]
+    - At larger sizes (>32 entries), they are implemented as a [hash array mapped trie][hamt] &#91;[1][stackoverflow]&#93;
+- [Maps][maps] can be declared with a literal form:
+
+  ```elixir
+  # An empty map
+  %{}
+
+  # A map with the atom key :a associated to the integer value 1
+  %{a: 1}
+
+  # A map with the string key "a" associated to the float value 2.0
+  %{"a" => 2.0}
+
+  # A map with the map key %{} with the list value [1,2,3]
+  %{%{}, [1,2,3]}
+  ```
+
+- The [Map module][map-module], included with the standard library, has many useful functions for using [maps][maps].
+
+  ```elixir
+  Map.get(%{a: 2}, :a)
+  # => 2
+  Map.get(%{a: 2}, :b)
+  # => nil
+  ```
+
+- [Maps][maps] implement the Enumerable protocol, allowing use of [Enum module][enum] functions
```

Comment:
Do the exercise's prerequisites ensure that the student knows what the Enumerable protocol is?

Reply:
I think I need to add it as an after point in the lists exercise.

But I think the details of protocols in elixir should be left until later.

Reply:
Agreed.